src/ica_api.c

```diff
@@ -363,7 +363,7 @@ unsigned int ica_open_adapter(ica_adapter_handle_t *adapter_handle)
 	}
 	if (*adapter_handle != -1) {
 		/* Test if character device is accessible. */
-		if (!ioctl(*adapter_handle, Z90STAT_STATUS_MASK, &status_mask)) {
+		if (!ioctl(*adapter_handle, ZCRYPT_STATUS_MASK, &status_mask)) {
```

Comment:
The only question would be what sense it really makes to perform this check here at all. 
First: Why should the handle not be accessible if it could be opened? 
Second: Even if it is 'usable' now, this does not tell if it is still usable the next time you perform a request. A sandbox may have closed the fil handle in the meantime. 
But I guess thats a more philosophic question....

Reply:
Yes, I had same thoughts. First I thought, Jörg forgot to change the code which evaluates the content of the array but after having a look onto this code I am wondering if we need this at all. Why not let a failure happen on the 'real' ioctl ZSECSENDCPRB ?
Another option would be to change this code to evaluate something usefull - for example count the nr of online CCA cards and based on this maybe do not even try to send down a ZSECSENDCPRB ...

Reply:
Yes, using MAX_ZDEV_CARDIDS_EXT is a good idea. It was just 64 before, so I first kept it similar. I'm wondering about the comment below the ioctl:
```
	/*
	 * Do not fail if crypto device driver is not loaded and CPACF is not
	 * available as the software fallback will still work without an adapter
	 * handle.
	 */
```
At least for libica-cex this is not true, but also libica has sw fallbacks disabled by default since quite some time. Maybe we have to rework this with another commit.